### PR TITLE
Add backend kickstart

### PR DESCRIPTION
**Pull request by Sharaf0**

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const express = require('express');
```
Comment (AmrAbdulrahman) [inline server/app.js]: We should think of the backend as two separate modules.
The backend which's the "http server" which serves the frontend application.
And the backend which is the API.
Usually, we always have those two backends isolated. (at least in the MEAN and similar stacks)

And in that case, either:
* the frontend only has a pointer 'api_server_endpoint' so it can consume remotely the APIs. or
* the backend of the frontend application encapsulates this information and creates a proxy for all the 'api' calls. and all api calls taking place on the frontend is just relative '/api/resource' calls. which then the backend server forwards to the real API backend.

So, in summary:
If you mean that this backend should be just the frontend application backend. (aka, http server) then it's very good! and it 
But if you meant for that backend to be the APIs backend, then
1. we first should move to a completely different project, different repo.
2. and then it shouldn't obviously serve any 'static' files, that would be the frontend server responsibility.

Comment (AmrAbdulrahman) [reply server/app.js]: I think a second step would be to apply the same idea and create a simple API server that exposes an API GET /say-hi that responds with a hello world message. hopefully not one of the following ("koko", "5ara", "kozbara", ...) 😆
